### PR TITLE
fix(payments): INT-3611 fixing where nonce value is going to be reloaded

### PR DESCRIPTION
**Pull request by roberbnd**
## What? [INT-4266](https://jira.bigcommerce.com/browse/INT-4266)
change where the nonce value is going to be reloaded

## Why?
the nonce value was being reloaded before be used then the value was empty, triggering one more time the popup to select the credit card
this happens because the backend delete the nonce value after be used: [Link](https://github.com/bigcommerce/bigcommerce/pull/39827/files#diff-ec498346f9c15c814cc5db89dd9abe81d9509baad8b0085124ed355238e145c8R438)

the function `loadPaymentMethod` calls the backend (deleting the nonce value) then I have to be careful where is called

## Testing / Proof
Happy path, it works form checkout and the cart section
https://drive.google.com/file/d/1vfEomqIo-Y_Ob0FYEMbXvl0yrTcZEh8x/view?usp=sharing

the original issue with the nonce value is fixed: [INT-3611](https://jira.bigcommerce.com/browse/INT-3611)
https://drive.google.com/file/d/1jF8xMUZkK_sInv_W2G5QNRELB-CtUaTB/view?usp=sharing

@bigcommerce/checkout @bigcommerce/payments @bigcommerce/apex-integrations

### CODE REVIEW

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts [hunk 1]
```diff
@@ -89,6 +85,11 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
 
         const { methodId } = payload.payment;
 
+        if (!this._paymentMethod?.initializationData.nonce) {
+            const state = this._store.getState();
+            this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-payment-strategy.ts:90]: 🍹 
```suggestion
            this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(methodId);
```

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts [hunk 2]
```diff
@@ -97,6 +98,10 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
                 onPaymentSelect = () => {},
             } = this._googlePayOptions;
             await this._displayWallet(methodId, onPaymentSelect, onError);
+
+            const state = this._store.getState();
+            this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-payment-strategy.ts:103]: 🍹 
```suggestion
            this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(methodId);
```

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts [hunk 3]
```diff
@@ -153,32 +158,31 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
...
             },
         };
     }
 
-    private _getNonce(methodId: string, { initializationData: { nonce }}: PaymentMethod) {
+    private async _getNonce(methodId: string) {
+        if (!this._paymentMethod) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+        const nonce = this._paymentMethod.initializationData.nonce;
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-payment-strategy.ts:179]: 🍹 
```suggestion
        const { nonce } = this._paymentMethod.initializationData;
```

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts [hunk 4]
```diff
@@ -198,6 +202,16 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
         // TODO: Revisit how we deal with GooglePaymentData after receiving it from Google
         await this._googlePayPaymentProcessor.handleSuccess(paymentData);
 
+        const state = this._store.getState();
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-payment-strategy.ts:206]: 🍹 
```suggestion
        this._paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(methodId);
```

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts [hunk 5]
```diff
@@ -198,6 +202,16 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
...
+
+        if (this._paymentMethod.initializationData.nonce) {
+            return await Promise.all([
+                this._store.dispatch(this._checkoutActionCreator.loadCurrentCheckout()),
+                this._store.getState(),
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-payment-strategy.ts:211]: Why is this necessary? 🤔

File: src/payment/strategies/googlepay/googlepay-payment-strategy.ts [hunk 6]
```diff
@@ -198,6 +202,16 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
         // TODO: Revisit how we deal with GooglePaymentData after receiving it from Google
         await this._googlePayPaymentProcessor.handleSuccess(paymentData);
 
+        const state = this._store.getState();
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+
+        if (this._paymentMethod.initializationData.nonce) {
```
Comment (mauricio-sg) [inline src/payment/strategies/googlepay/googlepay-payment-strategy.ts:208]: Keep consistency with line 181 or use this way in 181 😬

**Review by mauricio-sg — CHANGES_REQUESTED** [review]
Don't forget to run a regression 😱

**Comment by roberbnd** [comment]
it continues with this new PR  https://github.com/bigcommerce/checkout-sdk-js/pull/1122
